# Задания по исключениям.
# """
# """
# 1) Опишите полный синтаксис конструкции try-except:
# """
# SyntaxError
# IndentationError # ошибки с пробелом/ отступом
# TabError
# ZeroDivisionError
# ArithmeticError
# NameError
# IndexError
# KeyError
# ValueError
# BaseException # прородитель всех исключений

# shift + tab --> можно убрать оптом все отступы
# """
# 2) Дан следующий код:
# try:
#     b = 10
#     c = 11
#     print(a)
# except NameError:
#     print('Такой переснной не существует')

# """
# 3) Напишите программу которая будет получать два числа через input и делить одно на другое. Обработайте ошибку, которая возникнет в случае, если второе число окажется 0 и выведите сообщение.
# """
# try:
#     num1 = int(input('num : '))
#     num2 = int(input('num : '))
#     print(f'{num1} / {num2} = {num1 / num2}')
# except ZeroDivisionError: 
#     print('на ноль делить нельзя!')
# """
# 4) Напишите программу, которая будет получать через инпут 2 числа и будет печатать их сумму. Обработайте ошибку, которая возникнет, если пользователь введёт не числовое значение и выведите сообщение.
# """
# try:
#     num1 = int(input('num : '))
#     num2 = int(input('num : '))
#     print(f'{num1} + {num2} = {num1 + num2}')
# except ValueError:
#     print('Not number')
# """
# 5) Дан словарь. Попытайтесь получить значение по ключу. Обработайте ошибку, возникающую в том случае, если запрашиваемый ключ не существует.
# """
# try:
#     dict_ = {
#         1: 'bbir',
#         2: 'eki',
#         3: 'uch'
#     }
#     print(dict_[4])
# except KeyError:
#     print(' Nfrjuj rk.xf yt ceotcnetn')

# """
# 6)  Дан список. Обработайте исключение при попытке обращения к несуществующему элементу.
# """
# ls = list(range(1, 10))
# try:
#     print(ls[10])
# except IndexError:
#     print('Такого значения не существует.')
# """
# 7) Попытайтесь в блоке try принять 2 числа и разделить одно на другое. В блоке except обработайте сразу 2 возможных исключения.
# """
# num1 = int(input('num : '))
# num2 = int(input('num : '))
# try:
#     print(f'{num1} / {num2} = {num1 / num2}')
# except ValueError:
#     print('Not number')
# except ZeroDivisionError:
#     print('на ноль делить нельзя!')

# """
# 8 Запросите у пользователя сумму, которая у него сейчас есть в бумажнике. Если он введёт сумму, меньшую чем 0, то выбросите исключение с текстом "Сумма не может быть отрицательной!"
# """
# num1 = input('сумму, которая у него сейчас есть в бумажнике: ')
# try:
#     if num1 < 0:
#         raise ValueError
# except:
#     print('Сумма не может быть отрицательной!')


# num1 = input('сумму, которая у него сейчас есть в бумажнике: ')
# try:
#     if num1 >= 0:
#         print('ok')
#     else:
#         raise ValueError
# except:
#     print('Сумма не может быть отрицательной!')

# """
# 9)  В блоке try запросите у пользователя ввод его возраста. Затем в том же блоке проверьте его возраст на совершеннолетие. Если пользователь несовершеннолетний, выбросите исключение с текстом "Доступ запрещён". Обработайте также исключение если пользователь вводит текст вместо числа с сообщением 'Введен некорректный возраст'. Если ошибок не возникло, то распечатайте сообщение "Спасибо!", а затем "До свидания", вне зависимости от того, произошла ошибка или нет.
# """
# try:
#     age = int(input('Enter age: '))
#     if age < 18:
#         print('Доступ запрещён')
# except ValueError:
#     print('Введен некорректный возраст')
# else:
#     print('Спасибо!')
# finally:
#     print('До свидания')


